dojo/tools/blackduck/importer.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from pathlib import Path
+from typing import Iterable, Mapping, Set, Tuple
+from dojo.tools.blackduck.model import BlackduckFinding
+import csv
+import io
+import zipfile
+
+
+class Importer(ABC):
+    @abstractmethod
+    def parse_findings(self, report: Path) -> Iterable[BlackduckFinding]:
+        while False:
+            yield None
+
+
+class BlackduckImporter(Importer):
+    def parse_findings(self, report: Path) -> Iterable[BlackduckFinding]:
+        files = dict()
+        security_issues = dict()
+        try:
+            if zipfile.is_zipfile(report):
+                print("This is a zip file")
+                with zipfile.ZipFile(report) as zip:
+                    for file_name in zip.namelist():
+                        if file_name.endswith('files.csv'):
+                            print("dealing with files")
+                            with io.TextIOWrapper(zip.open(file_name), newline='') as f:
+                                files = self.__partition_by_project_id(f)
+                        elif file_name.endswith('security.csv'):
+                            print("dealing with security")
+                            with io.TextIOWrapper(zip.open(file_name), newline='') as f:
+                                security_issues = self.__partition_by_project_id(f)
+            else:
+                print("Not a zip file?")
+                print(report)
+                # raise ValueError
+        except Exception as e:
+            print("Could not process zip file: {}".format(e))
+
+        project_ids = set(files.keys()) & set(security_issues.keys())
+        for project_id in project_ids:
+            locations = set()
+            for file_entry in files[project_id]:
+                path = file_entry[8]
+                archive_context = file_entry[9]
+                if archive_context:
+                    locations.add(f"{archive_context}{path[1:]}")
```

Comment:
```suggestion
                    locations.add("{}{}".format(archive_context, path[1:]))
```

Literal format strings are a 3.6 feature.